ts/query.go

```diff
@@ -0,0 +1,431 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Matt Tracy (matt@cockroachlabs.com)
+
+package ts
+
+import (
+	"container/heap"
+	"sort"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/util"
+)
+
+// normalizedData is used to normalize an InternalTimeSeriesData object for
+// use in a dataSpan.  This is accomplished by computing a constant offset
+// adjustment which adjusts each Sample offset to be relative to the start time
+// of the dataSpan, rather than the start time of the InternalTimeSeriesData.
+type normalizedData struct {
+	*proto.InternalTimeSeriesData
+	offsetAdjustment int32
+}
+
+// offsetAt returns the normalized offset for the Sample at the supplied index
+// in the InternalTimeSeriesData's Samples collection.
+func (rdc *normalizedData) offsetAt(idx int) int32 {
+	return rdc.Samples[idx].Offset + rdc.offsetAdjustment
+}
+
+// dataSpan is used to construct a normalized, monolithic view of a single time
```

Comment:
by "normalize"  you mean that you measure everything relative to the start time, right? When I hear normalization, I usually think of dividing by something.
What about monolithic? That it spans multiple sources? That sounds more like "aggregated".